src/NimBLEUUID.cpp

```diff
@@ -318,4 +320,34 @@ std::string NimBLEUUID::toString() {
     return ble_uuid_to_str(&m_uuid.u, buf);
 } // toString
 
+
+NimBLEUUID & NimBLEUUID::operator=(const NimBLEUUID & other) {
+    m_uuid.u.type = other.m_uuid.u.type;
+    if(other.m_valueSet) {
+        memcpy(m_uuid.u128.value, other.m_uuid.u128.value, 16);
```

Comment:
Maybe we should put a `swtich(other.m_uuid.type) {` then add memcpy for `u16`  and `u32` types as well.

Reply:
I had the same idea in the first place, but the difference is (at most) two extra compares to save 14 or 12 bytes of memcpy. I think this is all the same...

Reply:
We don’t need to memcpy all 3 values, just the one defined by the type value. Also it is needed for ‘ble_uuid_cmp()’ and ‘ble_uuid_to_str()’ to work correctly.